packages/xstate-wallet/src/workflows/application.ts

```diff
@@ -0,0 +1,94 @@
+import {MachineConfig, Machine, StateMachine, send} from 'xstate';
+import {FINAL, MachineFactory} from '@statechannels/wallet-protocols';
+import {
+  CreateChannelEvent,
+  OpenChannelEvent
+} from '@statechannels/wallet-protocols/src/protocols/wallet/protocol';
+import {CreateChannel, JoinChannel} from '@statechannels/wallet-protocols/lib/src/protocols';
+import {IStore} from '@statechannels/wallet-protocols/src/store';
+import {State, getChannelId} from '@statechannels/nitro-protocol';
+import {SendStates} from '@statechannels/wallet-protocols/src/wire-protocol';
+
+// Events
+type OpenEvent = CreateChannelEvent | OpenChannelEvent;
+
+interface PlayerStateUpdate {
+  type: 'PLAYER_STATE_UPDATE';
+  state: State;
+}
+
+export type ApplicationWorkflowEvent = PlayerStateUpdate | SendStates | OpenEvent;
+
+// Context
+interface ApplicationContext {
+  channelId: string;
+}
+// States
+const initializing = {on: {CREATE_CHANNEL: 'funding', OPEN_CHANNEL: 'funding'}};
+
+const funding = {
+  invoke: {
+    id: 'openMachine',
+    src: 'invokeOpeningMachine',
+    data: (context, event) => event,
+    onDone: 'running',
+    autoForward: true
+  },
+  on: {SendStates: {actions: ['updateStore', 'sendChannelUpdated']}}
+};
+const running = {
+  on: {
+    PLAYER_STATE_UPDATE: {target: 'running', actions: ['sendToOpponent']},
+    SendStates: {target: 'running', actions: ['updateStore', 'sendChannelUpdated']}
+  }
```

Comment:
It might be worth writing out the logic of when you move to `'closing'` etc, with a mocked `CloseChannel` service. That would let us test starting a second game.

Reply:
Going to handle this in a separate PR for https://github.com/statechannels/monorepo/issues/899